[TabBar] Add new TabBar component.

kofi-clever opened the pull request:
Basically a row-direction FlexBox with buttons/anchors as child items.

![tabbar-reprise](https://cloud.githubusercontent.com/assets/16216084/19752530/f054a214-9bb2-11e6-9880-013cfee74e0b.gif)

CarlNelson commented on docs/TabBarExample.less:
```diff
@@ -0,0 +1,46 @@
+
+.TabBarExample {
+  .margin--top--2xl;
+  font-size: 0.875rem;
```
We don't have a text--size for this, but I see it used semi-often. I wonder if we should make one or just enforce size--medium or size--small and remove this in between.

kofi-clever replied:
true, might be worth adding a size definition for it. we do use .875rem and 14px quite a bit, but not firmly on one side or the other.
was thinking about this recently; 16px sometimes feels a tad too large for web app UI text. feels less of an issue for documentation/preamble text or for copy-heavy pages, but slightly off inside UI elements/controls. 

WDYT? i'm aware of my bias toward what i've used on other projects, so i'd defer to you on this one :)
definitely seen a trend toward larger in the last few years, but again that seems to be more focused on content-centric sites vs web apps.

CarlNelson replied:
I tend to fall on the make the text bigger camp for legibility purposes even though I tend to prefer very small text sizes for myself.

10 / 12 / 14 / 16 feels like too many close type sizes to me and doesn't fit a scale very well. Will think on it.

CarlNelson commented on src/less/border.less:
```diff
@@ -0,0 +1,6 @@
+@borderWidthLarge: .1875rem; // 3px
+
+@borderRadiusSmall: .125rem; // 2px
+@borderRadiusMedium: .1875rem; // 3px
```
I wonder if we should be more explicit about these similar to how we do the spacing file. Thoughts?

kofi-clever replied:
ah, you mean generate classes for `border--top--s`, `borderRadius--topLeft--m`, etc?
that could be useful. i'll take a look and add a commit

CarlNelson replied:
👍

kofi-clever commented on docs/index.html:
```diff
@@ -2,7 +2,6 @@
 <html>
     <head>
         <script src="bundle.js" type="text/javascript" charset="utf-8"></script>
-        <link rel="stylesheet" type="text/css" href="https://s3.amazonaws.com/assets.clever.com/design/style.css">
```
this gets in the way of development a lot, because it pulls in outdated styles that interfere with the new ones we're testing. no need to have it here since all the demo components are pulling in the styles they need already (we just end up with duplicate styles in the browser).

kofi-clever commented:
Ok, added auto-generated border definitions in https://github.com/Clever/components/pull/68/commits/0c0218bd04ff90b96430a614cd2d05f96adb6671 

Added classes for `border-radius`, but leaving `border` as mixins for now, since it wouldn't be too useful (it would require defaulting to a specific color that would have to be changed in css anyway)

kofi-clever commented:
rebased with changes in master - ready for review now

kofi-clever commented on docs/TabBarExample.less:
```diff
@@ -0,0 +1,46 @@
+  }
+
+  h2 {
+    .type_large;
```
thanks - updated

kofi-clever commented on genBorder.js:
```diff
@@ -0,0 +1,70 @@
+  s: 0.0625, // 1px
+  m: 0.125, // 2px
+  l: 0.1875, // 3px
+};
```
yeah, was thinking that too, but tested increasing the base font-size to see the effects and the borders growing along with the text actually didn't seem out of place at all. ends up working like an actual zoom.

that being said, not sure if that's actually what we want. seems fine, but will let @CarlNelson weigh in

CarlNelson replied:
If the world were only ever made of rem and em units I would be happier. The ability for rem/em to scale more clearly is one of the parts that I think pixels don't do as well.
